src/plugins/data_source/server/saved_objects/data_connection_saved_objects_client_wrapper.ts

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  SavedObjectsBulkCreateObject,
+  SavedObjectsBulkResponse,
+  SavedObjectsBulkUpdateObject,
+  SavedObjectsBulkUpdateOptions,
+  SavedObjectsBulkUpdateResponse,
+  SavedObjectsClientWrapperFactory,
+  SavedObjectsClientWrapperOptions,
+  SavedObjectsCreateOptions,
+  SavedObjectsUpdateOptions,
+  SavedObjectsUpdateResponse,
+} from 'opensearch-dashboards/server';
+import { SavedObjectsErrorHelpers } from '../../../../../src/core/server';
+import {
+  DATA_CONNECTION_ID_LENGTH_LIMIT,
+  DATA_CONNECTION_SAVED_OBJECT_TYPE,
+  DataConnectionType,
+} from '../../common/data_connections';
+
+/**
+ * Describes the Saved Objects Client Wrapper class,
+ * which contains the factory used to create Saved Objects Client Wrapper instances
+ */
+export class DataConnectionSavedObjectsClientWrapper {
+  /**
+   * Describes the factory used to create instances of Saved Objects Client Wrappers
+   * for data connection specific operations
+   */
+  public wrapperFactory: SavedObjectsClientWrapperFactory = (wrapperOptions) => {
+    const create = async <T = unknown>(
+      type: string,
+      attributes: T,
+      options?: SavedObjectsCreateOptions
+    ) => {
+      if (DATA_CONNECTION_SAVED_OBJECT_TYPE !== type) {
+        return await wrapperOptions.client.create(type, attributes, options);
+      }
+
+      this.validateAttributes(attributes);
+
+      return await wrapperOptions.client.create(type, attributes, options);
```

Comment:
simplify this to 
```
if (type === ...) {
    this.validaeAttributes(...);
}

return await wrapperOptions.client.create(type, attributes, options);
```

Reply:
Removing the wrapper file for now, to introduce the right checks in a follow-up PR.